controllers/nutanixmachine_controller_test.go

```diff
@@ -467,3 +469,519 @@ func TestNutanixMachineReconciler_SetupWithManager_ClusterToTypedObjectsMapperEr
 	assert.Error(t, err)
 	assert.Contains(t, err.Error(), "failed to create mapper for Cluster to NutanixMachine")
 }
+
```

Comment:
I first adapted these ginkgo tests for #492, but when I needed more flexibility in setup/teardown, I rewrote them using the standard "table test" pattern: https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/492/files#diff-6a4ea53969c034e97ecf2934960f526e89fa8f2aef3819b25c2d19531ee69673

Once #492 merges, I can backport my tests here.